src/App.tsx

```diff
@@ -1,26 +1,315 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { TodoForm } from './components/TodoForm';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { getPreparedTodos } from './utils/todoFilter';
+import { Filter } from './types/Filter';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [filterBy, setFilterBy] = useState(Filter.All);
+
+  const [titleInput, setTitleInput] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isSubmitting]);
+
+  const handleTitileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleInput(event.target.value);
+  };
+
+  const completedTasks = todos.filter(todo => todo.completed);
+
+  const todoCount = todos.length - completedTasks.length;
+
+  const handleClearCompleted = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    if (completedTodos.length === 0) {
+      return;
+    }
+
+    setIsSubmitting(true);
+
+    const deletePromises = completedTodos.map(todo =>
+      todoService
+        .deleteTodos(todo.id)
+        .then(() => todo.id)
+        .catch(() => {
+          setErrorMessage(prev => prev || 'Unable to delete a todo');
+
+          return null;
+        }),
+    );
+
+    Promise.allSettled(deletePromises).then(results => {
+      const successfullyDeletedIds = results
+        .filter(
+          (result): result is PromiseFulfilledResult<number | null> =>
+            result.status === 'fulfilled' && result.value !== null,
+        )
+        .map(result => result.value as number);
+
+      setTodos(currentTodos =>
+        currentTodos.filter(todo => !successfullyDeletedIds.includes(todo.id)),
+      );
+
+      setIsSubmitting(false);
+    });
+  };
+
+  const preparedTodos = getPreparedTodos(todos, filterBy);
+
+  function loadTodos() {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
  const loadTodos = () => {
```